pager.c

```diff
@@ -286,13 +287,20 @@ static const struct Mapping PagerNewsHelp[] = {
     break;                                                                     \
   }
 
-#define CHECK_READONLY                                                         \
-  if (!Contex2 || Contex2->mailbox->readonly)                                  \
-  {                                                                            \
-    mutt_flushinp();                                                           \
-    mutt_error(_(Mailbox_is_read_only));                                       \
-    break;                                                                     \
-  }
+
+inline static bool assert_mailbox_writable(struct Mailbox *mailbox)
+{
+  assert(mailbox);
+  if (mailbox->readonly)
+  {
+    mutt_flushinp();
+    mutt_error(_(Mailbox_is_read_only));
+    return false;
+  } 
+  else
+    return true;
```

Comment:
We can drop the `else` here (slightly flattening the code).
If the `if` statement is true, the function ends.

Reply:
done